src/lib/Liblog/pbs_log.c

```diff
@@ -884,16 +893,16 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 			snprintf(microsec_buf, sizeof(microsec_buf), ".%06ld", (long)tp.tv_usec);
 	}
 
+	/* lock the log mutex */
```

Comment:
moving the mutex log before the call to localtime actually is not required. It will not make anything any better. If there is a fork that happens and takes the localtime_r's internal mutex to deadlock, this will not prevent it. The original place for the mutex is actually fine, i.e. it only guards against the switching of the log files